modules/ossm-directing-egress-traffic-through-a-gateway-using-istio-apis.adoc

```diff
@@ -0,0 +1,226 @@
+----
+apiVersion: networking.istio.io/v1alpha3
+kind: Gateway
+metadata:
+  name: egress-gw
+  namespace: <gateway_namespace> # Namespace where the egress gateway is deployed
+spec:
+  selector:
+    istio: <gateway_name> # Selects the egress-gateway instance to handle this traffic
+  servers:
+    - port:
+        number: 80
+        name: http
+        protocol: HTTP
+      hosts:
+        - docs.redhat.com # External service host, not a full URL.
+---
+apiVersion: networking.istio.io/v1alpha3
+kind: DestinationRule
+metadata:
+  name: egress-dr
+  namespace: <gateway_namespace> # Namespace where the egress gateway is deployed
+spec:
+  host: <gateway_name>.<gateway_namespace>.svc.cluster.local
+  subsets:
+    - name: rh-docs
+----
+
+. Create a YAML file named `http-vs.yaml` that sets up a `VirtualService` to manage the flow of traffic from the application sidecars through the egress gateway to the external host.
++
+.Example configuration
+[source,yaml,subs="attributes,verbatim"]
+----
+apiVersion: networking.istio.io/v1alpha3
+kind: VirtualService
+metadata:
+  name: egress-vs
+  namespace: curl # Namespace where the curl pod is running
+spec:
+  hosts:
+    - docs.redhat.com # External service host, not a full URL.
+  gateways:
+    - mesh
+    - <gateway_namespace>/egress-gw # Egress gateway name defined in the file that you used in the previous step.
+  http:
+    - match:
+        - gateways:
+            - mesh
+          port: 80
+      route:
+        - destination:
+            host: <gateway_name>.<gateway_namespace>.svc.cluster.local
+            subset: rh-docs
+            port:
+              number: 80
+          weight: 100
+    - match:
+        - gateways:
+            - <gateway_namespace>/egress-gw # Egress gateway name defined in the file that you used in the previous step.
+          port: 80
+      route:
+        - destination:
+            host: docs.redhat.com
+            port:
+              number: 80
+          weight: 100
+----
+
+. Apply the YAML file by running the following command:
++
+[source,terminal]
+----
+$ oc apply -f http-vs.yaml
+----
+
+. Resend the HTTP request to the URL:
++
+[source,terminal]
+----
+$ oc exec "$CURL_POD" -n curl -c curl -- curl -sSL -o /dev/null -D - http://docs.redhat.com
+----
++
+The terminal should display information similar to the following output:
++
+.Example output
+[source,terminal]
+----
+...
+HTTP/1.1 301 Moved Permanently
+...
+location: <example_url>
+...
+
+HTTP/2 200
+Content-Type: text/html; charset=utf-8
+----
+
+. Ensure that the request was routed through the gateway by running the following command:
++
+[source,terminal]
+----
+$ oc logs deployment/<gateway_name> -n <gateway_namespace> | tail -1
+----
++
+[NOTE]
+====
+Access logging must be enabled for this verification step to work. You can enable Access logging to the standard output by setting the `spec.values.meshConfig.accessLogFile` field to `/dev/stdout` in the {istio} resource.
```

Comment:
Does "Access" really have to be capitalized?

Reply:
done